dm/docs/RFCS/20220124_enhance_task_manageability.md

```diff
@@ -0,0 +1,129 @@
+# Proposal: Enhance task manageability
+
+- Author(s): [ehco](https://github.com/Ehco1996)
+- Last updated: 2022-01-22
+
+## Background
+
+The main purpose of this change is to address the root cause of problems like [#3771](https://github.com/pingcap/tiflow/issues/3771), which are caused by the fact that commands about DM task do not distinguish between dynamic configuration and static resources, making it impossible for users to intuitively manage their tasks. For this reasons, we will attempt to redesign the state machine of DM task and optimize the dmctl interaction interface to provide a better user experience.
+
+### Current State machine of task
+
+![dm old task lifecycle](../media/dm-old-task-lifecycle.png)
+
+### New State machine of task
+
+![dm new task lifecycle](../media/dm-new-task-lifecycle.png)
+
+This is mainly achieved by adding a new command `dmctl task create`, which creates a stopped task instead of creating and starting a new task with single command `dmctl start-task`.
+
+## Goals
+
+- Organize and optimize the state machine of tasks for better management
+- Unified style of commands in dmctl, optimized dmctl interaction experience
+- Adding/modifying OpenAPI, enhance eco-tools
+
+## Design and Examples
+
+New syntax of dmctl is `dmctl [resource type] [command] [flags]`
+
+where `resource type`, `command` and `flags` are:
+
+- `resource type`: Specifies the resource you want to control. resource types are case-insensitive and there are limited resource types. currently there are only these types of resource types: `task`, `source`, `relay`, `ddl-lock` and `member`.
+
+- `command`: Specifies the operation that you want to perform on one or more resources, for example `create`, `get`, `update`, `delete`, `etc`.
+
+- `flags`: Specifies optional flags. For example, you can use the `--master-addr` flags to specify the address and port of the DM-Master server. Note that we **not allow** any non-keyword arguments, All arguments must be specified as this: `--flag value`.
```

Comment:
since a explicit call would be more intuitive than an implicit call, so i think `--remove-meta=ture`  is better than `--meta`?

wait for other reviewers cc @lance6716

Reply:
ping @lance6716 😬

Reply:
I'm more likely to prefer the former one. But `--remove-meta=true` looks a bit redundant because setting `--remove-meta` flag is sufficiently intuitive to express the semantics that users would like to remove the metadata (nobody would think setting this flag is about to preserve the metadata😂) . `--remove-meta` would be better.

Reply:
u are right, we should keep `--remove-meta` beacuse some paramters like`remove-meta` can have default value
but i still want to remove support for non-keyword args 

e.g:
```
dmctl task start --task-name="test" --remove-meta   ✔
dmctl task start test --remove-meta   ❌
```

Reply:
cc @sunzhaoyang . The object of a command does not need to be put in a flag IMO.

Reply:
agree.  I believe "task start [--some-flags] <object>" is better. We should not mix flags and object.

Reply:
remove the change now, patl @lance6716 

https://github.com/pingcap/tiflow/pull/4458/commits/11b6487444036d0f4033ab91dbfc376d4b2e5910